avocado/core/loader.py

```diff
@@ -694,8 +693,7 @@ def ignore_broken(klass, uid, params=None):
                                                 subtests_filter)
             else:
                 if os.access(test_path, os.X_OK):
-                    return self._make_test(test.SimpleTest,
-                                           pipes.quote(test_path))
+                    return self._make_test(test.SimpleTest, test_path)
```

Comment:
OK this little guy is actually very important, because we do support 2 ways of specifying test-ids. One is a simple path, which means it does not require escaping spaces. Then, for simple tests, we also allow providing path&args, which means the spaces inside location have to be escaped, because we do the shlex.split (line 499 in loaders).

Anyway our `utils.process` library, in case shell=False, also does the `shlex.split` so if we don't escape the url in (1) it splits the single name.

Words are words, let's talk in practice:

```
touch "foo bar"
chmod +x "foo bar"
avocado run "foo\ bar baz"  => executes SimpleTest "foo bar" with argument "baz" (path 2)
avocado run "foo bar" => executes SimpleTest "foo bar" (path 1)
```
but with this patch:
```
avocado run "foo\ bar baz"  => executes SimpleTest "foo\ bar baz" (path 2), which executes ["foo bar", "baz]
avocado run "foo bar" => executes SimpleTest "foo bar" (path 1), which executes ["foo", "bar"]
```

One thing I don't understand is why do you remove the `pipes.quote`, when I return it back it works well, the selftest passes and I'm able to execute the tests properly.